### PR TITLE
use folders in CMake

### PR DESCRIPTION
**Pull request by Dimon4eg**
This sets folders for targets to be visible in an IDE.

Before in VS (if tests ore on):
NC_3 2

Now in VS:
NC_3 2_folders

### CODE REVIEW

File: CMakeLists.txt
Comment (mgallien) [inline CMakeLists.txt:3]: Please could you add it to `cmake/modules/DefineCMakeDefaults.cmake` ?

**Review by mgallien — CHANGES_REQUESTED** [review]
Thanks, nice idea
Please address my comment and we should good to go

**Comment by Dimon4eg** [comment]
@mgallien take a look again plz

**Comment by mgallien** [comment]
Thanks
Please do not merge master when working on a PR but rebase on top of master. We will always rebase before merging a PR and the history should not include merge like this one. I have to write some documentation about that to make contributing easier.

**Review by mgallien — APPROVED** [review]

**Comment by Dimon4eg** [comment]
I see, I just pressed button `Update branch` in this PR

**Comment by allexzander** [comment]
> I see, I just pressed button `Update branch` in this PR

No worries. For the future, you will need to add a comment:

/rebase

in the PR. This will do the trick.

**Review by allexzander — APPROVED** [review]

**Comment by Dimon4eg** [comment]
`use folders for nextcloudsync source files` commit makes this:

image

**Comment by Dimon4eg** [comment]
`use folders for nextcloudCore source files` commit makes this:

image

**Review by mgallien — CHANGES_REQUESTED** [review]
Please next time, once a PR is approved do not push more changes to it. That is not fair for the reviewers.

Please remove your new changes and push them to another PR. It would be nice if we can merge your first one.

**Comment by Dimon4eg** [comment]
Ok, sorry, no problem

**Review by mgallien — APPROVED** [review]

**Pull request merged**
